### PR TITLE
Add latest and earliest to datetime fingerprints

### PR DESCRIPTION
**Pull request by sbelak**
Having this information allows us to be smart in determining the default time granularity and such.

### CODE REVIEW

**Comment by salsakran** [comment]
As we've learned over and over, the only sane way to store dates is something that includes a timezone =)

File: src/metabase/sync/analyze/fingerprint/datetime.clj [hunk 1]
```diff
@@ -0,0 +1,17 @@
+(ns metabase.sync.analyze.fingerprint.datetime
+  "Logic for generating a `DateTimeFingerprint` from a sequence of values for a `:type/DateTime` Field."
+  (:require [metabase.sync.interface :as i]
+            [schema.core :as s]))
+
+(s/defn datetime-fingerprint :- i/DateTimeFingerprint
+  "Generate a fingerprint containing information about values that belong to a `DateTime` Field."
+  [values :- i/FieldSample]
+  (let [[value & values] values]
+    (zipmap [:earliest :latest]
```
Comment (salsakran) [inline src/metabase/sync/analyze/fingerprint/datetime.clj]: is this really the simplest way to get a min/max or a set of values?

Comment (salsakran) [reply src/metabase/sync/analyze/fingerprint/datetime.clj]: or are we doing this for performance reasons?

**Comment by sbelak** [comment]
Depends/probably not. I’m more and more leaning towards using timestamps. That solves the timezone pitfalls and makes this much more clean & clear.

File: src/metabase/sync/analyze/fingerprint/datetime.clj (hunk 1 above)
Comment (senior) [reply src/metabase/sync/analyze/fingerprint/datetime.clj]: Maybe to avoid two passes over the data. Are there going to be a huge amount of dates here?

**Review by senior — COMMENTED** [review]
I don't think comparing the strings as dates will work out. Storing the datetime as millis won't work well either as it will assume UTC and we don't always know the timezone of the data to be able to convert it. I think it'll need to be a ISO timestamp string with the timezone if we have it. We'll need to parse it like we do query parameters and compare it using the objects.

**Comment by sbelak** [comment]
Did a less ghetto version using proper datetime arithmetics.

**Pull request merged**